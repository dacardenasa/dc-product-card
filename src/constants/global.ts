export const noImageURL = "https://image-test-aws.s3.us-west-1.amazonaws.com/no-image.jpg"