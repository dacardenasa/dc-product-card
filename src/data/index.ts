import { noImageURL } from "../constants";

export const product1 = {
  id: '1',
  title: 'Cofee Mug - sin imagen',
  // uriImage: "/coffee-mug.png"
};

export const product2 = {
  id: '1',
  title: 'Cofee Mug - CardById',
  uriImage: noImageURL,
};
