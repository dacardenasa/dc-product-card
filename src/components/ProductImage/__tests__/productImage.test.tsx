import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage } from '..';
import { ProductCard } from '../../ProductCard';
import { noImageURL } from '../../../constants';
import { product2 } from '../../../data';

describe('ProductImage', () => {
  test('render correctlly component', () => {
    const wrapper = renderer.create(
      <ProductImage uriImage={noImageURL} />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('render component without uriImage prop correctlly', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
