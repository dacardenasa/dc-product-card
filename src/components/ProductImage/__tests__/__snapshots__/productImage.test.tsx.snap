// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage render component without uriImage prop correctlly 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="product-image"
    className="productImg undefined"
    src="https://image-test-aws.s3.us-west-1.amazonaws.com/no-image.jpg"
  />
</div>
`;

exports[`ProductImage render correctlly component 1`] = `
<img
  alt="product-image"
  className="productImg undefined"
  src="https://image-test-aws.s3.us-west-1.amazonaws.com/no-image.jpg"
/>
`;
